### PR TITLE
Update to use latest Connector SDK

### PR DESCRIPTION
**Pull request by sirwolfgang**
This introduces a new Javascript API that can be used instead of or with the DOM API, giving exposure to exit events. There are a few ways to use it:

## 1. HTML5

If you're using the HTML interface, and need to upgrade to using some Javascript code, you can; but all event registrations are on a global level. This means that if you have multiple buttons, you will look at the metadata of the response to see which one you're reacting to.

```html
<head>
  <script src="http://cdn.lvh.me:3000/v1/connector.js"></script>
  <script language="JavaScript">
    Quiltt.onExitSuccess((metadata) =>
      console.log("Global onExitSuccess", metadata.connectionId)
    );
  </script>
</head>
<body>
  <button quiltt-button="<CONNECTOR_ID">Click Here!</button>
</body>
```

## 2. Javascript

Now if you want to do something more complex, and expect to be working with multiple buttons in different ways, then the Javascript SDK may be the way to go. With this, you can control everything in JS.

```html
<head>
  <script src="http://cdn.lvh.me:3000/v1/connector.js"></script>
  <script language="JavaScript">
    Quiltt.authenticate("<SESSION_TOKEN>");

    const connector = Quiltt.connect("<CONNECTOR_ID>", {
      onExitSuccess: (metadata) =>
        console.log("Connector onExitSuccess", metadata.connectionId),
    });

    connector.open();
  </script>
</head>
```

## 3. React

With these new hooks, the React components now support callbacks.

```tsx
import { QuilttButton } from '@quiltt/react'

export const App = () => {
  const [connectionId, setConnectionId] = useState<string>()
  const handleSuccess = (metadata) => setConnectionId(metadata?.connectionId)

  return (
    <QuilttButton connectorId="<CONNECTOR_ID>" onExitSuccess={handleSuccess}>
      Add
    </QuilttButton>

    <QuilttButton connectorId="<CONNECTOR_ID>" connectionId={connectionId}>
      Repair
    </QuilttButton>
  )
}
```

## Todos:
- [x] fix url in example and changeset 
- [x] Add checklist check to branch protections

### CODE REVIEW

File: ECMAScript/react/src/hooks/useQuilttConnector.ts [hunk 1]
```diff
@@ -1,39 +1,97 @@
...
 
-  // Create Script Element
+  // Set Session
   useEffect(() => {
-    if (scriptElement) return
+    if (typeof Quiltt === 'undefined') return
 
-    scriptElement = document.createElement('script')
-    scriptElement.src = `${QUILTT_CDN_BASE}/v1/connector.js`
+    Quiltt.authenticate(session?.token)
```
Comment (rubendinho) [inline ECMAScript/react/src/hooks/useQuilttConnector.ts:25]: @sirwolfgang should this run even if there is no token in the session?

**Review by rubendinho — APPROVED** [review]
LGTM. I pushed up some copy fixes, and made some updates to the root README.

File: .changeset/breezy-books-visit.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+</body>
+```
+
+## 2. Javascript
+
+Now if you want to do something more complex, and expect to be working with multiple buttons in different ways, then the Javascript SDK may be the way to go. With this, you can control everything in JS.
+
+```html
+<head>
+  <script src="http://cdn.lvh.me:3000/v1/connector.js"></script>
```
Comment (zubairaziz) [inline .changeset/breezy-books-visit.md:33]: Should this example show the production CDN?

**Review by zubairaziz — APPROVED** [review]

File: .changeset/breezy-books-visit.md (hunk 2 above)
Comment (rubendinho) [reply .changeset/breezy-books-visit.md:33]: Good catch - we need to fix it. Thanks

File: ECMAScript/react/src/hooks/useQuilttConnector.ts (hunk 1 above)
Comment (sirwolfgang) [reply ECMAScript/react/src/hooks/useQuilttConnector.ts:25]: Define "no" token, `undefined` shouldn't do anything, `null` should clear.

Comment (rubendinho) [reply ECMAScript/react/src/hooks/useQuilttConnector.ts:25]: Ok got it. If I understand correctly, The session has a token or there is no session, so null isn't expected.

**Review by tom-quiltt — APPROVED** [review]

**Pull request merged**